pr_agent/git_providers/github_provider.py

```diff
@@ -26,6 +26,8 @@ def set_pr(self, pr_url: str):
         self.pr = self._get_pr()
 
     def get_files(self):
+        if hasattr(self.pr, 'files'):
```

Comment:
Consider using a getter method to access the 'files' attribute of 'pr' object. Directly accessing or modifying object attributes is not a good practice as it breaks the encapsulation principle of OOP. [important]